packages/BetterPhpDocParser/PhpDocNodeVisitor/ChangedPhpDocNodeVisitor.php

```diff
@@ -20,8 +21,25 @@ public function beforeTraverse(Node $node): void
     public function enterNode(Node $node): ?Node
     {
         $origNode = $node->getAttribute(PhpDocAttributeKey::ORIG_NODE);
-        if ($origNode === null) {
+        if (! $origNode instanceof Node) {
             $this->hasChanged = true;
+            return $node;
+        }
+
+        $nodeStartAndEnd = $node->getAttribute(PhpDocAttributeKey::START_AND_END);
```

Comment:
`$nodeStartAndEnd` somehow got `null` value on `PhpDocInfo::removeByType()` called with remove `$this->markAsChanged()` call.

Removing this part:

https://github.com/rectorphp/rector-src/blob/f37a6eb0de44306065b24137a7922190622f5ccc/packages/BetterPhpDocParser/PhpDocInfo/PhpDocInfo.php#L300

make `e2e/applied-rule-change-docblock` error:

```bash
➜  applied-rule-change-docblock git:(compare-start-and-end) ✗ php ../e2eTestRunner.php
 3/3 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%    ---------- begin diff ----------
@@ @@
-1 file with changes
-===================
+2 files with changes
+====================

-1) src/RenameDocblock.php:0
+1) src/UselessVarTag.php:1

     ---------- begin diff ----------
 @@ @@
+
+ final class UselessVarTag
+ {
+-    /**
+-     * @var string
+-     */
+     public string $name = 'name';
+ }
+    ----------- end diff -----------
+
+Applied rules:
+ * RemoveUselessVarTagRector
+
+
+2) src/RenameDocblock.php:0
+
+    ---------- begin diff ----------
+@@ @@
  <?php

  /**
@@ @@
  * RenameClassRector


- [OK] 1 file would have changed (dry-run) by Rector
+ [OK] 2 files would have changed (dry-run) by Rector
```